frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/k8s-watch-types.ts

```diff
@@ -0,0 +1,20 @@
+import { Dispatch } from 'react-redux';
+import { K8sModel } from '../../../api/common-types';
+import { WatchK8sResource } from '../../../extensions/console-types';
+
+export type RootState = any;
```

Comment:
Need to not use `RootState` in our SDK, we have a type for this and we need to infer/extend or use it directly...

```
diff --git a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/k8s-watch-types.ts b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/k8s-watch-types.ts
index 39981fb86a..b43e8ce983 100644
--- a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/k8s-watch-types.ts
+++ b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/k8s-watch-types.ts
@@ -1,13 +1,12 @@
 import { Dispatch } from 'react-redux';
 import { K8sModel } from '../../../api/common-types';
+import { SDKStoreState } from '../../../app/redux-types';
 import { WatchK8sResource } from '../../../extensions/console-types';

-export type RootState = any;
-
-export type GetIDAndDispatch = (
+export type GetIDAndDispatch = <StoreData extends SDKStoreState>(
   resource: WatchK8sResource,
   k8sModel: K8sModel,
-) => { id: string; dispatch: (dispatch: Dispatch, getState: () => RootState) => void };
+) => { id: string; dispatch: (dispatch: Dispatch, getState: () => StoreData) => void };

 export type Query = { [key: string]: any };

diff --git a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResource.ts b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResource.ts
index fe9c087201..ffb7d3d679 100644
--- a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResource.ts
+++ b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResource.ts
@@ -4,11 +4,11 @@ import { Map as ImmutableMap } from 'immutable';
 // @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
 import { useSelector, useDispatch } from 'react-redux';
 import { K8sModel } from '../../../api/common-types';
+import { SDKStoreState } from '../../../app/redux-types';
 import * as k8sActions from '../../../app/k8s/actions/k8s';
 import { getReduxIdPayload } from '../../../app/k8s/reducers/k8sSelector';
 import { UseK8sWatchResource } from '../../../extensions/console-types';
 import { getReference } from '../k8s-ref';
-import { RootState } from './k8s-watch-types';
 import { getIDAndDispatch, getReduxData, NoModelError } from './k8s-watcher';
 import { useDeepCompareMemoize } from './useDeepCompareMemoize';
 import { useModelsLoaded } from './useModelsLoaded';
@@ -21,7 +21,7 @@ export const useK8sWatchResource: UseK8sWatchResource = (initResource) => {
     ? getReference(resource.groupVersionKind)
     : resource?.kind;

-  const k8sModel = useSelector<RootState, K8sModel>(({ k8s }) =>
+  const k8sModel = useSelector<SDKStoreState, K8sModel>(({ k8s }) =>
     kindReference ? k8s.getIn(['RESOURCES', 'models', kindReference]) : null,
   );

@@ -40,7 +40,7 @@ export const useK8sWatchResource: UseK8sWatchResource = (initResource) => {
     };
   }, [dispatch, reduxID]);

-  const resourceK8s = useSelector<RootState, ImmutableMap<string, any>>((state) =>
+  const resourceK8s = useSelector<SDKStoreState, ImmutableMap<string, any>>((state) =>
     reduxID ? getReduxIdPayload(state, reduxID.id) : null,
   );

diff --git a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResources.ts b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResources.ts
index 797d2548e5..d8f79dce4e 100644
--- a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResources.ts
+++ b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useK8sWatchResources.ts
@@ -7,9 +7,10 @@ import { createSelectorCreator, defaultMemoize } from 'reselect';
 // import * as k8sActions from '@console/internal/actions/k8s';
 import { K8sModel } from '../../../api/common-types';
 import * as k8sActions from '../../../app/k8s/actions/k8s';
+import { SDKStoreState } from '../../../app/redux-types';
 import { UseK8sWatchResources } from '../../../extensions/console-types';
 import { getReference } from '../k8s-ref';
-import { GetIDAndDispatch, RootState } from './k8s-watch-types';
+import { GetIDAndDispatch } from './k8s-watch-types';
 import { getIDAndDispatch, getReduxData, NoModelError } from './k8s-watcher';
 import { useDeepCompareMemoize } from './useDeepCompareMemoize';
 import { useModelsLoaded } from './useModelsLoaded';
@@ -19,8 +20,8 @@ export const useK8sWatchResources: UseK8sWatchResources = (initResources) => {
   const resources = useDeepCompareMemoize(initResources, true);
   const modelsLoaded = useModelsLoaded();

-  const allK8sModels = useSelector<RootState, ImmutableMap<string, K8sModel>>((state: RootState) =>
-    state.k8s.getIn(['RESOURCES', 'models']),
+  const allK8sModels = useSelector<SDKStoreState, ImmutableMap<string, K8sModel>>(
+    (state: SDKStoreState) => state.sdkK8s.getIn(['RESOURCES', 'models']),
   );

   const prevK8sModels = usePrevious(allK8sModels);
@@ -101,7 +102,7 @@ export const useK8sWatchResources: UseK8sWatchResources = (initResources) => {
   const resourceK8sSelector = React.useMemo(
     () =>
       resourceK8sSelectorCreator(
-        (state: RootState) => state.sdkK8s,
+        (state: SDKStoreState) => state.sdkK8s,
         (sdkK8s) => sdkK8s,
       ),
     [resourceK8sSelectorCreator],
diff --git a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useModelsLoaded.ts b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useModelsLoaded.ts
index 5188e4b6af..c0d1a2ebb8 100644
--- a/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useModelsLoaded.ts
+++ b/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/useModelsLoaded.ts
@@ -3,15 +3,15 @@ import * as React from 'react';
 // @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
 import { useSelector } from 'react-redux';
 import { K8sModel } from '../../../api/common-types';
-import { RootState } from './k8s-watch-types';
+import { SDKStoreState } from '../../../app/redux-types';

 export const useModelsLoaded = (): boolean => {
   const ref = React.useRef(false);
-  const k8sModels = useSelector<RootState, K8sModel>(({ k8s }) =>
-    k8s.getIn(['RESOURCES', 'models']),
+  const k8sModels = useSelector<SDKStoreState, K8sModel>(({ sdkK8s }) =>
+    sdkK8s.getIn(['RESOURCES', 'models']),
   );
-  const inFlight = useSelector<RootState, K8sModel>(({ k8s }) =>
-    k8s.getIn(['RESOURCES', 'inFlight']),
+  const inFlight = useSelector<SDKStoreState, K8sModel>(({ sdkK8s }) =>
+    sdkK8s.getIn(['RESOURCES', 'inFlight']),
   );

   if (!ref.current && k8sModels.size && !inFlight) {
```

Reply:
Updated the diff to use the Generic the right way (sorry about that) 🙂